test_suite/package/test_otel.py

```diff
@@ -0,0 +1,132 @@
+
+    assert host.backend.run_local(modify_iam_role_cmd), 'faild to update iam role'
+
+
+@pytest.fixture()
+def install_packages(request, host):
+    class_instance = request.node.cls
+    install_cmd = f'yum install -y  {class_instance.package_name} --nogpgcheck'
+    repo_path = ('https://copr.fedorainfracloud.org/coprs/miyunari'
+                 '/redhat-opentelemetry-collector/repo/rhel-9/miyunari-redhat-opentelemetry-collector-rhel-9.repo')
+    with host.sudo():
+        assert host.run_test(f'yum-config-manager --add-repo {repo_path}')
+        assert host.run_test(install_cmd), f'Failed to install the package {class_instance.package_name}'
+        output = host.check_output("rpm -qa | grep opentelemetry*")
+        print(output)
+
+
+@pytest.fixture()
+def start_service(request, host):
+    class_instance = request.node.cls
+    start_enable_service = (f'systemctl start {class_instance.package_name} && '
+                            f'systemctl enable {class_instance.package_name}')
+    is_active = (f'systemctl is-active {class_instance.package_name}')
+
+    with host.sudo():
+        assert host.run(start_enable_service).succeeded, (f'Failed to start the service {class_instance.package_name}')
+        assert host.run(is_active).succeeded, (f'Failed to activate the service {class_instance.package_name}')
+
+    def finalizer():
+        logger.info(f'Removing the package {class_instance.package_name}')
+        assert host.run(f'sudo yum remove -y {class_instance.package_name}')
+        assert not host.check_output(f'rpm -q {class_instance.package_name}')
+        assert not host.run(f'ssh {class_instance.instance_dns}').succeeded
+        logger.info("Verify logs don't appear")
+        # TODO: verify logs don't appear.
+    request.addfinalizer(finalizer)
+
+
+@pytest.fixture(scope="function", autouse=True)
+def run_subscription_manager_auto(request, host, instance_data):
+    class_instance = request.node.cls
+    # Run the subscription manager auto test before any tests in this file
+    TestsSubscriptionManager.test_subscription_manager_auto(class_instance, host, instance_data)
+
+
+@pytest.mark.package
+@pytest.mark.run_on(['rhel9.1'])
+class TestOtel:
+    package_name = 'opentelemetry-collector-cloudwatch-config'
+
+    def check_aws_cli_logs(self, host):
+        command_to_run = [
+            'aws', 'logs', 'filter-log-events',
+            '--log-stream-names', 'testing-integrations-stream-emf',
+            '--filter-pattern', 'Invalid',
+            '--log-group-name', 'testing-logs-emf'
+        ]
+        run_aws_cli_cmd = ' '.join(command_to_run)
+
+        command_output = host.backend.run_local(run_aws_cli_cmd), 'Failed to get log output'
+        assert "Invalid" in command_output
+
+    @pytest.mark.usefixtures(
+        check_instance_status.__name__,
+        install_packages.__name__,
+        modify_iam_role.__name__,
+        start_service.__name__,
+    )
+    def test_otel(self, host, instance_data):
```

Comment:
I think this test case is overloaded with too many fixtures that could potentially be other test cases or change the running conditions:
* `run_subscription_manager_auto` should not be run every time a test case is run. This is only needed by `install_packages`.
* `install_packages` and  `modify_iam_role` could be fixtures to be run once, at a class level.
* `start_service` should be a test case that runs first. E.g. `test_otel_service`.
  * the `finalizer` should be another test case, that runs in the end. E.g. `test_no_logs_after_removing_otel`.

Reply:
To change the order in which the test cases are run, please use the `order(num)` marker.
You can find examples in other test cases from the generic or cloud-specific test suites.